### PR TITLE
Feature: Add federation results counts grouped by portal

### PR DESCRIPTION
**Pull request by Bilelkihal**
### Changes
- Add text for api json button in search page and show it only when the search is performed (https://github.com/ontoportal-lirmm/bioportal_web_ui/commit/f3aa7b4cef8cca09f826fc64e4e5801743537d2d)
Capture d’écran 2024-10-16 à 19 34 03
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -- - - - - - - - - -
Capture d’écran 2024-10-16 à 19 37 49

-----

- Show results count separately for every portal in federated search and browse (https://github.com/ontoportal-lirmm/bioportal_web_ui/commit/20d85ec46637f25fc9a233ae3351ba49b35e562d) (https://github.com/ontoportal-lirmm/bioportal_web_ui/commit/63f27eea2a129752e24b8e978f400684cef9f968) (https://github.com/ontoportal-lirmm/bioportal_web_ui/commit/76c1db5811ebf6e5134045b0b682ad5a903858ad) (https://github.com/ontoportal-lirmm/bioportal_web_ui/commit/caf73538976a163114a461f782f06a544e52f50d)
image
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -- - - - - - - - - -
image

-----

- Fix browse page federation portals button colors issue (https://github.com/ontoportal-lirmm/bioportal_web_ui/commit/e2074c72e4e05dcc560dd03afadfe37ac8f91542)
image

### CODE REVIEW

File: app/views/search/index.html.haml [hunk 1]
```diff
@@ -56,21 +59,21 @@
                 = t('search.hide_advanced_options')
     - unless @federation_errors.blank?
       = render Display::AlertComponent.new(message: @federation_errors, type: "warning")
-    - if @search_results
```
Comment (syphax-bouazzouni) [inline app/views/search/index.html.haml:59]: why removing this?

File: app/views/search/index.html.haml [hunk 2]
```diff
@@ -40,10 +40,13 @@
         .search-page-options{class:  @search_results.empty? ? 'justify-content-end': ''}
           - unless @search_results.empty?
             .search-page-number-of-results
-              = "#{t('search.match_in')} #{@search_results.length} #{t('search.ontologies')} #{t('federation.from')} #{request_portals_names.join(', ').html_safe} (#{sprintf("%.2f", @time)}s)".html_safe
+              = "#{t('search.match_in')} #{@search_results.length} #{t('search.ontologies')} #{t('federation.from')} #{request_portals_names(@federation_counts).join(', ').html_safe} in #{sprintf("%.2f", @time)}s".html_safe
```
Comment (syphax-bouazzouni) [inline app/views/search/index.html.haml:43]: here we need to extract the common code with browse page

File: app/views/search/index.html.haml (hunk 1 above)
Comment (Bilelkihal) [reply app/views/search/index.html.haml:59]: it has no sense, we have a loop `@search_results.each` in the next line (also `@search_results `can't be `nil` but `[]` so if I put it back we need to use `unless @search_results.blank?` instead

File: app/components/ontology_browse_card_component.rb [hunk 3]
```diff
@@ -17,7 +17,7 @@ def ontology
   end
 
   def external_ontology?
-    !internal_ontology?(@ontology[:id]) || (Array(@ontology[:sources]).size > 1)
```
Comment (syphax-bouazzouni) [inline app/components/ontology_browse_card_component.rb:20]: why removing this `|| (Array(@ontology[:sources]).size > 1)` ?

File: app/views/search/index.html.haml (hunk 1 above)
Comment (syphax-bouazzouni) [reply app/views/search/index.html.haml:59]: If not needed don't put it back

File: app/components/ontology_browse_card_component.rb (hunk 3 above)
Comment (Bilelkihal) [reply app/components/ontology_browse_card_component.rb:20]: The ontologies that are present in the current portal, we count them internal. 
I noticed that this change break one thing, which is the portals buttons when the ontology is internal and also present in other portals and I fixed it in the commit (https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/7db6a96f1bad0f3209ccc794174f1fd904c344d7)

File: app/views/search/index.html.haml (hunk 2 above)
Comment (Bilelkihal) [reply app/views/search/index.html.haml:43]: It's not the same sentence the only extractable thing is the name of the portals + the count which is already extracted in request_portals_names function

**Review by syphax-bouazzouni — APPROVED** [review]
Approved, and added the following changes:

- [use helpers for re-used ui components in the search UI page](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/247031742c85fe369723377aecc46466b89928cf)
- [extract same helper for the error message alert UI component](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/6c740e5d606c3e76a54d9cf3a929770a23794b16)
- [harmonize federation result text code](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/0559a3d9596d7b0239fb0017fa1ae3c2e59e99b4)
- [fix text_with_icon helper style](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/97f2f3f7800de38595b0d364df0677ac6511f3df)
- [remove no more used :from local text key](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/72fd27099c245ef0a4e2222bd8662eaa0b9a9693)
- [complete match_in local text](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/b8fa13b3a2b6b09b812ca320d9dd8ed5d5739fdb)
- [move urls related helpers from application_helper to urls_helpers file](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/d0f5ee07d0f54a9604de67c0c11742ad6f1ecee9)
- [update search aggregator and result component to save ontology id](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/a653fd1342cb941a8b0250e85e7c0a9a7f90819f)
- [extract common logic to count ontology id from search and browse results](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/ac1f608256472b59ed7b1c24cf3c2bd51657f70e)
- [clean url helpers usages](https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/a985763dc734b4e4648e1975c71c7aeda46b0f82)

File: app/components/ontology_browse_card_component.rb (hunk 3 above)
Comment (syphax-bouazzouni) [reply app/components/ontology_browse_card_component.rb:20]: okay

File: app/views/search/index.html.haml (hunk 2 above)
Comment (syphax-bouazzouni) [reply app/views/search/index.html.haml]: done in (https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/778/commits/0559a3d9596d7b0239fb0017fa1ae3c2e59e99b4)

**Pull request merged**